_inc/connect-button.js

```diff
@@ -52,7 +52,8 @@ jQuery( document ).ready( function ( $ ) {
 		handleConnectInPlaceFlow: function () {
 			// Alternative connection buttons should redirect to the main one for the "connect in place" flow.
 			if ( connectButton.hasClass( 'jp-banner__alt-connect-button' ) ) {
-				window.location = jpConnect.connectInPlaceUrl;
+				// Make sure we don't lose the `from` parameter.
+				window.location = jpConnect.connectInPlaceUrl + '&from=' + connectButtonFrom;
```

Comment:
Is there ever a situation where there won't be parameters on the URL? Should we transform to an URL and use URLSearchParams to append it?

Reply:
When I was playing with `URLSearchParams`, I noticed it didn't handle params within hashes, as it treats everything after the hash to be a... hash. 

That's a good point though, there is no reason to append anything at all if `connectButtonFrom` is empty at this point.

Reply:
@kraftbj @kbrown9 I took another stab at it with URLSearchParams, but it's not working for me. I think because `.append()` doesn't actually return anything? I assume the `undefined` check that's used elsewhere in the file is for graceful fallback for IE, so I left it in there. I probably won't have much time to re-visit this today, but please feel free to add it if you think it's worth it. Also I'd be curious to see what you had in mind. 

`if ( connectButton.hasClass( 'jp-banner__alt-connect-button' ) ) {
				// Make sure we don't lose the `from` parameter, if set.
				if ( 'undefined' === typeof URLSearchParams ) {
					var fromParam = ( connectButtonFrom && '&from=' + connectButtonFrom ) || '';
					window.location = jpConnect.connectInPlaceUrl + fromParam;
				} else {
					var searchParams = new URLSearchParams( jpConnect.connectInPlaceUrl );
					window.location = searchParams && searchParams.append( 'from', connectButtonFrom );
				}
				return;
			}`

And the `URLSearchParams` snippet to re-fetch the URL in the diff below (works): 

`if ( 'undefined' === typeof URLSearchParams ) {
		connectButtonFrom = location.hash.split( '&from=' )[ 1 ];
	} else {
		var searchParams = new URLSearchParams( location.hash.replace( /#\//, '' ) );
		connectButtonFrom = searchParams && searchParams.get( 'from' );
	}`

Reply:
Since we want this to land in 9.0, I am okay just landing this and following up. I need to check it out, but my understanding is `append` doesn't return, just adds to the object. You'll need to specifically ask for the object to be stringed for use again.